test/Bug356.fst

```diff
@@ -0,0 +1,31 @@
+module Bug356
+#lang-pulse
+open Pulse.Nolib
+
+fn test (b : bool)
+  returns x : UInt32.t
+{
+  if (b) {
+    1ul;
+  } else {
+    admit();
+    2ul;
+  }
+}
+
+fn test_final_admit (b : bool)
+  returns x : UInt32.t
+{
+  if (b) {
+    1ul;
+  } else {
+    admit();
+  }
+}
+
+fn test_unit_admit (x: bool)
+{
+  admit ();
+  let _ = test x;
+  ()
+}
```

Comment:
This `admit ()` causes the body of the function to be silently dropped during extraction.

That's because `admit #unit ()` is extracted to unit, which is usually a good thing because it's how we admit proofs.